org/mozilla/jss/ssl/javax/JSSParameters.java

```diff
@@ -0,0 +1,174 @@
+        }
+
+        // Copy data from downcast
+        setCipherSuites(downcast.getCipherSuites());
+        setProtocols(downcast.getProtocols());
+        setAlgorithmConstraints(downcast.getAlgorithmConstraints());
+        setEndpointIdentificationAlgorithm(downcast.getEndpointIdentificationAlgorithm());
+        setNeedClientAuth(downcast.getNeedClientAuth());
+        setServerNames(downcast.getServerNames());
+        setSNIMatchers(downcast.getSNIMatchers());
+        setUseCipherSuitesOrder(downcast.getUseCipherSuitesOrder());
+        setWantClientAuth(downcast.getWantClientAuth());
+    }
+
+    public JSSParameters(String[] cipherSuites) {
+        super(cipherSuites);
+        setCipherSuites(cipherSuites);
+    }
+
+    public JSSParameters(String[] cipherSuites, String[] protocols) {
+        super(cipherSuites, protocols);
+        setCipherSuites(cipherSuites);
+        setProtocols(protocols);
+    }
+
+    public void setCipherSuites(String[] cipherSuites) throws IllegalArgumentException {
+        if (cipherSuites == null || cipherSuites.length == 0) {
+            suites = null;
+            return;
+        }
+
+        ArrayList<SSLCipher> converted = new ArrayList<SSLCipher>();
+        for (String cipherSuite : cipherSuites) {
+            try {
+                converted.add(SSLCipher.valueOf(cipherSuite));
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Unknown cipher suite (" + cipherSuite + "): " + e.getMessage(), e);
+            }
+        }
+
+        suites = converted.toArray(new SSLCipher[0]);
+    }
+
+    public void setCipherSuites(AbstractList<SSLCipher> cipherSuites) {
+        setCipherSuites(cipherSuites.toArray(new SSLCipher[0]));
+    }
+
+    public void setCipherSuites(SSLCipher[] cipherSuites) {
+        if (cipherSuites == null || cipherSuites.length == 0) {
+            suites = null;
+            return;
+        }
+
+        // Perform a copy of cipherSuites in case it is modified later.
+        suites = new SSLCipher[cipherSuites.length];
+        for (int index = 0; index < cipherSuites.length; index++) {
+            suites[index] = cipherSuites[index];
+        }
+    }
+
+    public String[] getCipherSuites() {
+        if (suites == null) {
+            return null;
+        }
+
+        ArrayList<String> ciphers = new ArrayList<String>();
+        for (SSLCipher suite : suites) {
+            ciphers.add(suite.name());
+        }
+
+        return ciphers.toArray(new String[0]);
+    }
+
+    public SSLCipher[] getSSLCiphers() {
+        return suites;
+    }
+
+    public void setProtocols(String[] protocols) {
+        if (protocols == null || protocols.length == 0) {
+            range = null;
+            return;
+        }
+
+        try {
+            SSLVersion minProtocol = SSLVersion.findByAlias(protocols[0]);
+            SSLVersion maxProtocol = SSLVersion.findByAlias(protocols[0]);
+
+            for (String protocol : protocols) {
```

Comment:
Just a slight redundancy, the protocols[0] will be compared by itself.

Reply:
Yeah; it is easier than handling `null` in `{min,max}Protocol` or starting `protocols` from index 1. There's also what, 6 protocols?

`SSLv2`, `SSLv3`, `TLSv1.0`, `TLSv1.1`, `TLSv1.2`, and `TLSv1.3`

So anything else is really overkill and this is provided for compatibility with the official standards; in TomcatJSS, we'll likely construct from either a `SSLVersion` pair or `SSLVersionRange` directly.